config/objects/generic.json

```diff
@@ -72,7 +75,9 @@
 					"zoneLimit"	: 1,
 					"value"		: 100,
 					"rarity"	: 100
-				}
+				},
+				"modes" : ["resource-resource", "resource-player"],
+				"efficacy" : 5
```

Comment:
Why "efficacy" and not "efficiency"? I think latter is more correct english)

Reply:
it's not about english, both words are correct, but have different meanings.
It's better to explain on synonyms: if for markets you prefer "productivity" then "efficiency".
If "performance" or "impact" then "efficacy".

BTW, it appears in that PR https://github.com/vcmi/vcmi/pull/2100

Reply:
Or use more direct meaning.
pricesFactor / costsFacts / exchangeRate / etc
To clearly indicate what this factor refers to

Reply:
BuildingSubID::BROTHERHOOD_OF_SWORD:
		b = createBonus(building, Bonus::MORALE, +2);

This will be in jsons in the future?

Reply:
And this part: "dwellingUpLvl1", BuildingID::DWELL_LVL_1_UP },

This is too rigid.

A creature building could be:
Base - normal creature building
Upgraded - upgraded creature building
Alternate - some other creature building to be chosen instead of base one.
Alternate upgrade - upgrade for the above
3rd upgrade - a second upgrade for the building,  HotA archers like.

Reply:
> This will be in jsons in the future?

this change makes possible to configure special town buildings and map objects in the same way.
But I didn't rewrite legacy json